src/controller/LottoController.js

```diff
@@ -0,0 +1,105 @@
+import LottoGenerator from '../domains/LottoGenerator';
+import LottoCalculator from '../domains/LottoCalculator';
+import InputView from '../views/InputView';
+import OutputView from '../views/OutputView';
+import LottoPaymentValidator from '../validators/LottoPaymentValidator';
+import LottoValidator from '../validators/LottoValidator';
+import executeWithRetry from '../utils/executeWithRetry';
+import LOTTO_RULES from '../constants/lotto-rules';
+
+class LottoController {
+  #lottoNumbers = {};
+
+  constructor() {
+    this.#lottoNumbers = {
+      winningNumbers: null,
+      bonusNumber: null,
+    };
+  }
+
+  async run() {
+    const tickets = await this.initTicketCount();
+    const lottoGenerator = await this.initLottoGenerator(tickets);
+    await this.initLottoNumbers();
+    await this.initLottoCalculator(tickets, lottoGenerator);
+    await this.reStartLotto();
+  }
+
+  async initTicketCount() {
+    const tickets = await this.readLottoPayment();
+    return this.getTicketCount(tickets);
```

Comment:
`tickets`라는 변수가 여기도 사용되는데, 다른 의미로 사용되고 있네요. 의미를 명확하게 표현하는 이름을 지어주시면 더 좋을 것 같아요.

```suggestion
    const lottoPayment = await this.readLottoPayment();
    return this.getTicketCount(lottoPayment);
```